test/helpers/getPerformanceTests.ts

```diff
@@ -0,0 +1,100 @@
+import testFs from './testFs';
+import {
+	evaluateXPathToNodes,
+	evaluateXPathToString,
+	evaluateXPathToFirstNode,
+} from '../../src/index';
+
+import {
+	ALL_TESTS_QUERY,
+	getAllTestSets,
+	getFile,
+	unrunnableTestCasesByName,
+} from './qt3TestsTools';
+
+const PERFORMANCE_TESTS_FILE = 'runnablePerformanceTestNames.csv';
+const TOTAL_NUMBER = 6420;
+let numberOfTest = 50;
+
+const createNewRunnableTests = () => {
+	const runningPossibilityForEachTest = TOTAL_NUMBER / numberOfTest;
+	const allTestSets = getAllTestSets();
+	const newRunnableTests = new Set();
+
+	function addTestNames() {
+		for (const testSetFileName of allTestSets) {
+			if (newRunnableTests.size >= numberOfTest) {
+				break;
+			}
+			const testSet = getFile(testSetFileName);
+
+			// Find all the tests we can run
+			const testCases = evaluateXPathToNodes(ALL_TESTS_QUERY, testSet);
+
+			if (!testCases.length) {
+				return;
+			}
+
+			for (const testCase of testCases) {
+				if (newRunnableTests.size >= numberOfTest) {
+					break;
+				}
+
+				try {
+					const testName = evaluateXPathToString('./@name', testCase);
+
+					if (unrunnableTestCasesByName[testName]) {
+						continue;
+					}
+
+					const assertionNode = evaluateXPathToFirstNode('./result/*', testCase);
+					const assertionType = assertionNode && (assertionNode as any).localName;
+
+					switch (assertionType) {
+						case 'assert':
+						case 'assert-true':
+						case 'assert-eq':
+						case 'assert-deep-eq':
+						case 'assert-empty':
+						case 'assert-false':
+						case 'assert-count':
+						case 'assert-type':
+						case 'assert-xml':
+						case 'assert-string-value':
+						case 'all-of':
+						case 'any-of':
+							if (Math.floor(Math.random() * runningPossibilityForEachTest) === 1) {
+								newRunnableTests.add(testName);
+							}
+							break;
+					}
+				} catch (e) {
+					console.error(e);
+					continue;
+				}
+			}
+		}
+	}
+
+	while (newRunnableTests.size < numberOfTest) {
+		addTestNames();
+	}
+
+	testFs.writeFileSync(PERFORMANCE_TESTS_FILE, [...newRunnableTests].join('\n'));
+};
+
+export default function getPerformanceTests() {
+	const regenerateIndex = process.argv.indexOf('regenerate');
+
+	if (regenerateIndex !== -1) {
+		const newNumber = Number(process.argv[regenerateIndex + 1]);
+		numberOfTest =
+			typeof newNumber === 'number' && !isNaN(newNumber) && newNumber > 0
```

Comment:
let's just throw if we get anything else than a number. And assume the last argument is the number we want

Reply:
done

Reply:
Do we throw the error elsewhere or did we decide to go for the default value instead of raising an error?

Reply:
I think going for the default value would be better

Reply:
I disagree, If it's not a number than the arguments are incorrect. I think it makes more sense to let the user know the arguments are incorrect instead of giving the impression that nothing is wrong.

Reply:
I usually prefer erroring instead of sweeping mistakes under the carpet, but I do not care too much either way